### PR TITLE
Compiler Requirements for Branch Nodes fix

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Computing requirements for compilation missed the elseNode for branches.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [X] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/659

### CODE REVIEW

File: pkg/compiler/test/compiler_test.go [hunk 1]
```diff
@@ -109,21 +109,25 @@ func TestBranches(t *testing.T) {
 		}
 
 		t.Run(path, func(t *testing.T) {
-			//if !strings.HasSuffix(path, "success_5.json") {
+			//if !strings.HasSuffix(path, "success_6.json") {
```
Comment (kumare3) [inline pkg/compiler/test/compiler_test.go:112]: huh?!

Comment (kumare3) [reply pkg/compiler/test/compiler_test.go:112]: what happened to the previous test case?

**Review by kumare3 — APPROVED** [review]
minor comment, LGTM!

**Review by wild-endeavor — APPROVED** [review]

File: pkg/compiler/test/compiler_test.go (hunk 1 above)
Comment (wild-endeavor) [reply pkg/compiler/test/compiler_test.go:112]: this is just commented out... it's to skip the irrelevant test cases - a test was added so it was bumped.

**Review by bnsblue — APPROVED** [review]
rubber stamping it just for merging. it is already reviewed by 2+ contributors.

**Comment by EngHabu** [comment]
Thank you @bnsblue

**Pull request merged**